### PR TITLE
Pyseir location cleanup + start of removing county_metadata

### PR DESCRIPTION
**Pull request by ghop02**
A couple things this PR accomplishes:
 * removes all state level data access functions in load_data - everything is accessed through fips.
 * Start removing most of the county_metadata references.  County metadata is another source that goes outside of our traditional data pipeline.  Removing the references helps streamline some of the pyseir code.  That source has always been very opaque to me.
 * One of the confusing things is that we use the full state name in a lot of locations when we should be using fips or the ISO 2 abbreviated name.  I start to try to make it more explicit where we do that in this PR.  I'd like to take another pass through in a separate PR to finish that cleanup.
 * It seems we only really use county metadata in the ensemble_runner for age distribution.  Do we actually use that? it would be great if we could find another way to save that data.

### CODE REVIEW

File: pyseir/inference/initial_conditions_fitter.py [hunk 1]
```diff
@@ -229,6 +229,7 @@ def _fit_fips(fips, generate_report=False):
         return {"model_params": None, "t0_date": None, "reduced_chi2": None}
 
 
+# Chris: Can we delete??? Not used anywhere and I don't think start dates matter...
```
Comment (BrettBoval) [inline pyseir/inference/initial_conditions_fitter.py:232]: Remove this whole file (InitialConditionsFitter class) and the cli calls associated with this. It isn't used anymore.

**Comment by BrettBoval** [comment]
> * It seems we only really use county metadata in the ensemble_runner for age distribution.  Do we actually use that? it would be great if we could find another way to save that data

Which line are you referencing with this?

**Comment by ghop02** [comment]
> > * It seems we only really use county metadata in the ensemble_runner for age distribution.  Do we actually use that? it would be great if we could find another way to save that data
> 
> Which line are you referencing with this?

https://github.com/covid-projections/covid-data-model/blob/master/pyseir/ensembles/ensemble_runner.py#L247-L254

**Comment by ghop02** [comment]
@BrettBoval I'm going to leave deleting the other code for another PR if that's alright!

**Comment by BrettBoval** [comment]
> > > * It seems we only really use county metadata in the ensemble_runner for age distribution.  Do we actually use that? it would be great if we could find another way to save that data
> > 
> > 
> > Which line are you referencing with this?
> 
> https://github.com/covid-projections/covid-data-model/blob/master/pyseir/ensembles/ensemble_runner.py#L247-L254

I don't know of anywhere where we use that data.

**Comment by BrettBoval** [comment]
> @BrettBoval I'm going to leave deleting the other code for another PR if that's alright!

That's fine with me.

**Pull request merged**